tokenmagic/module/tokenmagic.js

```diff
@@ -1925,7 +1925,68 @@ Hooks.on("preUpdateMeasuredTemplate", async (document, options) => {
   }
 });
 
-/* -------------------------------------------- */
+Hooks.on("updateMeasuredTemplate", (scene, data, options) => {
+  //log("Hook -> updateMeasuredTemplate");
+  if ( isNewerVersion(game.version, "0.8") ) {
+    [data, options] = [scene, data];
+    scene = scene.parent;
+  }
+
+  if ( scene.id !== game.user.viewedScene ) return;
+
+  let placeable = getPlaceableById(data._id, PlaceableType.TEMPLATE);
+  if ( options.hasOwnProperty("texture") ) {
+    // removing animations on this placeable
+    Anime.removeAnimation(data._id);
+
+    // clearing the filters (owned by tokenmagic)
+    Magic._clearImgFiltersByPlaceable(placeable);
+
+    // querying filters reload (when pixi containers are ready)
+    requestLoadFilters(placeable, 250);
+
+  }
+  else {
+    if ( !placeable.loadingRequest ) {
+      Magic._updateFilters(data, options, PlaceableType.TEMPLATE);
+      Magic._updateTemplateData(data, options, PlaceableType.TEMPLATE);
+    }
+  }
+});
+
+Hooks.on("deleteMeasuredTemplate", (parent, doc, options, userId) => {
+  //log("Hook -> deleteMeasuredTemplate");
+  if ( !(doc == null || !doc.hasOwnProperty("_id")) ) {
+    Anime.removeAnimation(doc._id);
+  }
+});
+
+Hooks.on("createMeasuredTemplate", (scene, data, options) => {
+  //log("Hook -> createMeasuredTemplate");
+  if ( isNewerVersion(game.version, "0.8") ) {
+    [data, options] = [scene, data];
+    scene = scene.parent;
+  }
+
+  if ( !(scene == null)
+    && scene.id === game.user.viewedScene
+    && data.hasOwnProperty("flags")
+    && data.flags.hasOwnProperty("tokenmagic")
+    && data.flags.tokenmagic.hasOwnProperty("filters") ) {
+
+    let placeable = getPlaceableById(data._id, PlaceableType.TEMPLATE);
+
+    // request to load filters (when pixi containers are ready)
+    requestLoadFilters(placeable, 250);
+  }
+});
+
+Hooks.on("preCreateMeasuredTemplate", (document, updateData) => {
+  // This would ideally be `preCreateMeasuredTemplate` and/or merged with the createMeasuredTemplate
+  const templates = TokenMagicSettings.getSystemTemplates();
+  if (templates?.enabled) {
+      templates.preCreateMeasuredTemplate?.(document);
+  }

```

Comment:
There is some old code here. Could you rebase your fork with the latest version?

Reply:
merged